sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -177,22 +179,45 @@ private ProgramState VisitInvocationExpression(InvocationExpressionSyntax instru
             private ProgramState ProcessStreamDisposingTypes(ProgramState programState, SyntaxNode usingExpression)
             {
                 var newProgramState = programState;
+                var objectCreationExpressions = usingExpression.DescendantNodes()
+                    .OfType<ObjectCreationExpressionSyntax>();
 
-                var arguments = usingExpression.DescendantNodes()
-                    .OfType<ObjectCreationExpressionSyntax>()
-                    .Where(IsStreamDisposingType)
-                    .Select(FirstArgumentOrDefault)
-                    .WhereNotNull();
-
-                foreach (var argument in arguments)
+                foreach (var objectCreationExpression in objectCreationExpressions)
                 {
-                    var streamSymbol = this.semanticModel.GetSymbolInfo(argument.Expression).Symbol;
-                    newProgramState = ProcessDisposableSymbol(newProgramState, argument.Expression, streamSymbol);
+                    var objectType = this.semanticModel.GetSymbolInfo(objectCreationExpression.Type)
+                    .Symbol
+                    .GetSymbolType();
+
+                    if (objectType.DerivesOrImplementsAny(typesDisposingUnderlyingStream) &&
+                        objectCreationExpression.ArgumentList?.Arguments is IEnumerable<ArgumentSyntax> argumentList)

```

Comment:
You could also check the  number of arguments, if it makes sense... Can you call the method with 2 arguments and still pass the required argument? (I'm not sure, so feel free to ignore the comment if not relevant)